### ISSUE TITLE
5.0.0-alpha.3: valid-jsdoc unexpectedly expects returns for async functions, even with requireReturn: false

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#10386]
**Tell us about your environment**

* **ESLint Version:** 5.0.0-alpha.3
* **Node Version:** 8.11.2
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "rules": {
  },
  "env": {
    "es6": true
  },
  "parserOptions": {
    "ecmaVersion": 9,
  },
  "rules": {
    "valid-jsdoc": ["error", {
      "prefer": {
	"return": "returns",
      },
      "requireReturn": false,
      "requireReturnDescription": false,
    }]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**                                                                                                                                                       
 * A comment.                                                                                                                                             
 */
async function foo() {
  if (!this.bar) {
    return;
  }

  await this.bar();
}
```

```bash
./node_modules/.bin/eslint eslint5.0test.js
```

**What did you expect to happen?**

No failures

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
eslint4.0test.js
  1:1  error  Missing JSDoc @returns for function  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)
```
This is a possible regression from #10161

**Comment by Standard8** [comment]
Now I'm having another look at #10161, I'm seeing that since that has been implemented, async functions must always have a @returns statement in the jsdoc, regardless of the `requireReturn` value.

For a function that is `async` the fact it returns a promise is implicit in its definition. Therefore, documentation wise, it could be treated the same as a function that returns nothing.

Unfortunately, I'm not quite seeing how making this optional could fit into the current configuration. 

So ideas here would be appreciated, /cc @not-an-aardvark @rachx

**Comment by rachx** [comment]
> For a function that is async the fact it returns a promise is implicit in its definition. Therefore, documentation wise, it could be treated the same as a function that returns nothing.

In the original issue, some users suggested that with `requireReturn: false`, it should also allow them to add jsdoc descriptions as `async` functions returns a promise. 

To make it optional for `async` function just like constructors and support both preferences, how about check if it is not an async function before reporting missing jsdocs?

https://github.com/eslint/eslint/blob/e0a0418fef214e0c548d2ddb02e98182e625f93a/lib/rules/valid-jsdoc.js#L407-L420

**Comment by Standard8** [comment]
@rachx now I re-read the docs again(!), I think having returns optional in the `requireReturn:false` case makes sense.

Updating that section seems to make sense as well. Do you want to do a PR for that, or I could do one tomorrow?

**Comment by rachx** [comment]
@Standard8 You can make the PR 👍
